website/content/en/pre-docs/tasks/running-pods.md

```diff
@@ -0,0 +1,222 @@
+So all key-value pairs must match if you use `nodeSelector`.
+Also, `nodeSelector` can do only do inclusions, while `affinity can do inclusions and exclusions (`In` and `NotIn`).
+Here is an example:
+
+```
+nodeSelector:
+  topology.kubernetes.io/zone: us-west-2a
+  node.k8.aws/capacity-type: spot
+  jones.dev: john
+```
+This example features a well-known label (`topology.kubernetes.io/zone`), a label that is specific to the AWS cloud provider (`node.k8.aws/capacity-type`), and a custom label (`jonees.dev`).
+Use a command like the following to set a custom label:
+
+```
+kubectl label nodes <node_name> <label_key>=<label_value>
+```
+
+See [nodeSelector](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector) in the Kubernetes documentation for details.
+
+To select a node by name, use the `nodeName` object:
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: myapp
+spec:
+  containers:
+  - name: myapp
+    image: myapp
+  nodeName: my-node
+```
+## Topology spread (`topologySpreadConstraints`)
+
+By using the Kubernetes `topologySpreadConstraints` you can ask the provisioner to have pods push away from each other for high availability reasons.
+Think of it as the Kubernetes evolution for pod affinity: it lets you relate pods with respect to nodes while still allowing spread.
+For example:
+
+```
+topologySpreadConstraints:
+  - maxSkew: 1
+    topologyKey: "topology.kubernetes.io/zone"
+    topologyKey: "topology.kubernetes.io/hostname"
+    whenUnsatisfiable: ScheduleAnyway
+```
+Adding this to your podspec would result in:
+
+* Pods being spread across both zones and hosts (`topologyKey`)
+* No more than one pod difference in the number of pods on each host (`maxSkew`).
+For example, if there were three nodes and five pods the pods could be spread 1, 2, 2 or 2, 1, 2 and so on.
+If instead the spread were 5, pods could be 5, 0, 0 or 3, 2, 0, or 2, 1, 2 and so on.
+* If the skew cannot be met, schedule the pods anyway (`whenUnsatisfiable`).
+
+See [Pod Topology Spread Constraints](https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/) for details.
+
+## Node affinity and anti-affinity (`nodeAffinity`)
+
+Examples below illustrate how to use Node affinity (`In`) and anti-affinity (`NotIn`).
+See [Node affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity) for details.
+When setting rules, the following Node affinity types define how hard or soft each rule is:
+
+* **requiredDuringSchedulingIgnoredDuringExecution**: This is a hard rule that must be met.
+* **preferredDuringSchedulingIgnoredDuringExecution**: This is a preference, but the pod can run on a node where it is not guaranteed.
+
+The `IgnoredDuringExecution` part of each tells the pod to keep running, even if conditions change on the node so the rules no longer matched.
```

Comment:
```suggestion
The `IgnoredDuringExecution` part of each tells the pod to keep running, even if conditions change on the node so the rules no longer matched. You can think of these concepts as `required` and `preferred`, since Kubernetes never implemented other variants of these rules.
```